http3/roundtrip.go

```diff
@@ -278,10 +278,35 @@ func isNotToken(r rune) bool {
 	return !httpguts.IsTokenRune(r)
 }
 
+func resolveUDPAddrWithCtx(ctx context.Context, netstr string, addr string) (*net.UDPAddr, error) {
+	var (
+		err        error
+		host, port string
+		portnum    int
+	)
+	if host, port, err = net.SplitHostPort(addr); err != nil {
+		return nil, err
+	}
+	if portnum, err = net.DefaultResolver.LookupPort(ctx, netstr, port); err != nil {
+		return nil, err
+	}
+	if host == "" {
+		return &net.UDPAddr{IP: net.IPAddr{}.IP, Port: portnum, Zone: net.IPAddr{}.Zone}, nil
+	}
+	ips, err := net.DefaultResolver.LookupIPAddr(ctx, host)
```

Comment:
I imagine that this will soon invite a request to use a different resolver here. How shall we deal with that?